src/dscanner/analysis/pointless_arithmetic_check.d

```diff
@@ -0,0 +1,165 @@
+import dscanner.analysis.base;
+import std.stdio;
+import std.typetuple;
+
+/**
+ * Checks for Pointless Arithmetic.
+ */
+final class PointlessArithmeticCheck : BaseAnalyzer
+{
+	alias visit = BaseAnalyzer.visit;
+
+	///
+	this(string fileName, bool skipTest = false)
+	{
+		super(fileName, null, skipTest);
+	}
+
+	override void visit(const AssignExpression asgExp)
+	{
+		auto left = asgExp.ternaryExpression;
+		UnaryExpression right = cast(UnaryExpression) asgExp.expression;
+
+		if (left && right)
+		{
+			if (asgExp.operator == tok!"*=" &&
+				isNumberLiteral(right.primaryExpression.primary.type) &&
+				right.primaryExpression.primary.text == "1")
+			{
+				addErrorMessage(asgExp.line, asgExp.column, KEY,
+								"Any number 'x' mul 1 will be 'x'.");
+			}
+			else if (asgExp.operator == tok!"/=" &&
+					 isNumberLiteral(right.primaryExpression.primary.type) &&
+					 right.primaryExpression.primary.text == "1")
+			{
+				addErrorMessage(asgExp.line, asgExp.column, KEY,
+								"Any number 'x' div 1 will be 'x'.");
+			}
+			else if (asgExp.operator == tok!"%=" &&
+					 isNumberLiteral(right.primaryExpression.primary.type) &&
+					 right.primaryExpression.primary.text == "1")
+			{
+				addErrorMessage(asgExp.line, asgExp.column, KEY,
+								"Any number modulo 1 will be 0.");
+			}
+		}
+		asgExp.accept(this);
+	}
+
+	override void visit(const PowExpression powExp)
+	{
+		UnaryExpression left = cast(UnaryExpression) powExp.left;
+		UnaryExpression right = cast(UnaryExpression) powExp.right;
+
+		if (left && right)
+		{
+			if (isNumberLiteral(right.primaryExpression.primary.type) &&
+				right.primaryExpression.primary.text == "0")
+			{
+				addErrorMessage(powExp.line, powExp.column, KEY,
+								"Any number pow 0 will be 1.");
+			}
+		}
+		powExp.accept(this);
+	}
+
+	override void visit(const MulExpression mulExp)
+	{
+		UnaryExpression left = cast(UnaryExpression) mulExp.left;
+		UnaryExpression right = cast(UnaryExpression) mulExp.right;
+
+		if (left && right)
+		{
+			if (mulExp.operator == tok!"*" &&
+				isNumberLiteral(right.primaryExpression.primary.type) &&
+				right.primaryExpression.primary.text == "1")
+			{
+				addErrorMessage(mulExp.line, mulExp.column, KEY,
+								"Any number 'x' mul 1 will be 'x'.");
+			}
+			else if (mulExp.operator == tok!"/" &&
+				isNumberLiteral(right.primaryExpression.primary.type) &&
+				right.primaryExpression.primary.text == "1")
+			{
+				addErrorMessage(mulExp.line, mulExp.column, KEY,
+								"Any number 'x' div 1 will be 'x'.");
+			}
+			else if (mulExp.operator == tok!"%" &&
+					 isIntegerLiteral(left.primaryExpression.primary.type) &&
+					 isNumberLiteral(right.primaryExpression.primary.type) &&
+					 right.primaryExpression.primary.text == "1")
+			{
+				addErrorMessage(mulExp.line, mulExp.column, KEY,
+								"Any number modulo 1 will be 0.");
+			}
+		}
+		mulExp.accept(this);
+	}
+
+	static bool isIntegerLiteral(IdType type) nothrow pure
```

Comment:
Please move this in the `private:` section. 
Also this could go in dparse in a further PR but for now it's ok as a local helper.